File: posts/mathworks-interview/1130.py
from typing import *

class Solution:
    def mctFromLeafValues(self, arr: List[int]) -> int:
        res = 0

        while len(arr) != 1:
            largestLeafValue = float("-inf")
            product = float("inf")
            index = -1

            for i in range(0, len(arr) - 1):
                a = arr[i]
                b = arr[i + 1]
                if a * b < product:
                    index = i
                    product = a * b
                    largestLeafValue = max(a, b)

            res += product
            arr[index: index + 2] = [largestLeafValue]

        return res

s = Solution()
print(s.mctFromLeafValues([6, 2, 4])) # 32

File: posts/mathworks-interview/1130.rs
struct Solution;

use std::cmp::max;
use std::i32;

impl Solution {
    pub fn mct_from_leaf_values(arr: Vec<i32>) -> i32 {
        let mut arr = arr.clone();
        let mut res: i32 = 0;

        while arr.len() != 1 {
            let mut largestLeafValue: i32 = i32::MIN;
            let mut product: i32 = i32::MAX;
            let mut index: i32 = -1;

            for i in 0..arr.len() - 1 {
                let a = arr[i];
                let b = arr[i + 1];
                if a * b < product {
                    largestLeafValue = max(a, b);
                    product = a * b;
                    index = i as i32;
                }
            }

            arr.remove(index as usize);
            arr.remove(index as usize);
            arr.insert(index as usize, largestLeafValue);
            res = res + product;
        }

        return res;
    }
}